From 8f22b88fe184e6cea34fd2029d5f210db2ec4a8f Mon Sep 17 00:00:00 2001
From: Jagan Teki <jagan@edgeble.ai>
Date: Wed, 2 Nov 2022 19:17:22 +0530
Subject: [PATCH] ARM: dts: rockchip: rv1126: Add USB Host, OTG node

Signed-off-by: Jagan Teki <jagan@edgeble.ai>
---
 arch/arm/boot/dts/rv1126.dtsi | 117 ++++++++++++++++++++++++++++++++++
 1 file changed, 117 insertions(+)

diff --git a/arch/arm/boot/dts/rv1126.dtsi b/arch/arm/boot/dts/rv1126.dtsi
index 1f07d0a4f..1e600aa1a 100644
--- a/arch/arm/boot/dts/rv1126.dtsi
+++ b/arch/arm/boot/dts/rv1126.dtsi
@@ -104,6 +104,15 @@ pmu_io_domains: io-domains {
 			status = "disabled";
 		};
 	};
+        qos_usb_host: qos@fe810000 {
+                compatible = "rockchip,rv1126-qos", "syscon";
+                reg = <0xfe810000 0x20>;
+        };
+
+        qos_usb_otg: qos@fe810080 {
+                compatible = "rockchip,rv1126-qos", "syscon";
+                reg = <0xfe810080 0x20>;
+        };
 
 	qos_emmc: qos@fe860000 {
 		compatible = "rockchip,rv1126-qos", "syscon";
@@ -170,6 +179,19 @@ power-domain@RV1126_PD_SDIO {
 				pm_qos = <&qos_sdio>;
 				#power-domain-cells = <0>;
 			};
+
+                        power-domain@RV1126_PD_USB {
+                                reg = <RV1126_PD_USB>;
+                                clocks = <&cru HCLK_USBHOST>,
+                                         <&cru HCLK_USBHOST_ARB>,
+                                         <&cru CLK_USBHOST_UTMI_OHCI>,
+                                         <&cru ACLK_USBOTG>,
+                                         <&cru CLK_USBOTG_REF>;
+                                pm_qos = <&qos_usb_host>,
+                                         <&qos_usb_otg>;
+                                #power-domain-cells = <0>;
+                        };
+
 		};
 	};
 
@@ -232,6 +254,45 @@ dmac: dma-controller@ff4e0000 {
 		clock-names = "apb_pclk";
 	};
 
+	u2phy0: usb2phy@ff4c0000 {
+		compatible = "rockchip,rv1126-usb2phy";
+		reg = <0xff4c0000 0x8000>;
+		rockchip,usbgrf = <&grf>;
+		clocks = <&pmucru CLK_USBPHY_OTG_REF>;
+		clock-names = "phyclk";
+		#clock-cells = <0>;
+		status = "disabled";
+
+		u2phy_otg: otg-port {
+			#phy-cells = <0>;
+			interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "otg-bvalid", "otg-id",
+					  "linestate";
+			status = "disabled";
+		};
+	};
+
+	u2phy1: usb2phy@ff4c8000 {
+		compatible = "rockchip,rv1126-usb2phy";
+		reg = <0xff4c8000 0x8000>;
+		rockchip,usbgrf = <&grf>;
+		clocks = <&pmucru CLK_USBPHY_HOST_REF>;
+		clock-names = "phyclk";
+		#clock-cells = <0>;
+		assigned-clocks = <&cru USB480M>;
+		assigned-clock-parents = <&u2phy1>;
+		clock-output-names = "usb480m_phy";
+		status = "disabled";
+
+		u2phy_host: host-port {
+			#phy-cells = <0>;
+			interrupts = <GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "linestate";
+			status = "disabled";
+		};
+	};
 	uart0: serial@ff560000 {
 		compatible = "rockchip,rv1126-uart", "snps,dw-apb-uart";
 		reg = <0xff560000 0x100>;
@@ -311,6 +372,62 @@ uart5: serial@ff5a0000 {
 		reg-io-width = <4>;
 		status = "disabled";
 	};
+	usbdrd: usb0 {
+		compatible = "rockchip,rv1126-dwc3", "rockchip,rk3399-dwc3";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		clocks = <&cru CLK_USBOTG_REF>, <&cru ACLK_USBOTG>,
+			 <&cru HCLK_PDUSB>;
+		clock-names = "ref_clk", "bus_clk", "hclk";
+		status = "disabled";
+
+		usbdrd_dwc3: dwc3@ffd00000 {
+			compatible = "snps,dwc3";
+			reg = <0xffd00000 0x100000>;
+			interrupts = <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>;
+			dr_mode = "otg";
+			maximum-speed = "high-speed";
+			phys = <&u2phy_otg>;
+			phy-names = "usb2-phy";
+			phy_type = "utmi_wide";
+			power-domains = <&power RV1126_PD_USB>;
+			resets = <&cru SRST_USBOTG_A>;
+			reset-names = "usb3-otg";
+			snps,dis_enblslpm_quirk;
+			snps,dis-u2-freeclk-exists-quirk;
+			snps,dis_u2_susphy_quirk;
+			snps,dis-del-phy-power-chg-quirk;
+			snps,tx-ipgap-linecheck-dis-quirk;
+			snps,tx-fifo-resize;
+			snps,xhci-trb-ent-quirk;
+			status = "disabled";
+		};
+	};
+
+	usb_host0_ehci: usb@ffe00000 {
+		compatible = "generic-ehci";
+		reg = <0xffe00000 0x10000>;
+		interrupts = <GIC_SPI 82 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru HCLK_USBHOST>, <&cru HCLK_USBHOST_ARB>,
+			 <&u2phy1>;
+		phys = <&u2phy_host>;
+		phy-names = "usb";
+		power-domains = <&power RV1126_PD_USB>;
+		status = "disabled";
+	};
+
+	usb_host0_ohci: usb@ffe10000 {
+		compatible = "generic-ohci";
+		reg = <0xffe10000 0x10000>;
+		interrupts = <GIC_SPI 83 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru HCLK_USBHOST>, <&cru HCLK_USBHOST_ARB>,
+			 <&u2phy1>;
+		phys = <&u2phy_host>;
+		phy-names = "usb";
+		power-domains = <&power RV1126_PD_USB>;
+		status = "disabled";
+	};
 
 	saradc: adc@ff5e0000 {
 		compatible = "rockchip,rv1126-saradc", "rockchip,rk3399-saradc";
-- 
2.40.1

